src/resolvers/Mutation/createPost.ts

```diff
@@ -93,6 +94,38 @@ export const createPost: MutationResolvers["createPost"] = async (
     imageUrl: args.file ? uploadImageObj?.newImagePath : "",
   });
 
+  // Check if the pinned property is set
+  if (args.data.pinned) {
+    // Check if the user has privileges to pin the post
+    const currentUser = await User.findOne({
```

Comment:
You're querying database two times, just to get current user's data. You've already done it before with currentUserExists. Instead of just using it to check whether user exists, fetch the user object, check its existence there and then also make use of its data here for comparision.

Reply:
Got it.